torchquad/integration/newton_cotes.py

```diff
@@ -47,8 +49,13 @@ def calculate_result(self, function_values, dim, n_per_dim, hs):
             backend tensor: Quadrature result
         """
         # Reshape the output to be [N,N,...] points instead of [dim*N] points
-        function_values = function_values.reshape([n_per_dim] * dim)
-
+        integrand_shape = function_values.shape[1:]
+        dim_shape = [n_per_dim] * dim
+        new_shape = [*integrand_shape, *dim_shape]
+        einsum = "".join([chr(i + 65) for i in range(len(function_values.shape))])
+        function_values = anp.einsum(f'{einsum}->{einsum[1:]}{einsum[0]}', function_values)
+        function_values = function_values.reshape(new_shape)
+        assert new_shape == list(function_values.shape)
```

Comment:
should we add a message explaining what happened if the assertion is wrong?

something like

`assert new_shape == list(function_values.shape), "function value reshaping filed due to (...)"` ?

Reply:
yup!